homeassistant/components/rituals_perfume_genie/number.py

```diff
@@ -0,0 +1,126 @@
+"""Support for Rituals Perfume Genie numbers."""
+from __future__ import annotations
+
+import logging
+
+from pyrituals import Diffuser
+
+from homeassistant.components.number import NumberEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import RitualsDataUpdateCoordinator
+from .const import ATTRIBUTES, COORDINATORS, DEVICES, DOMAIN, ROOM, SPEED
+from .entity import DiffuserEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_PERFUME_AMOUNT = 1
+MAX_PERFUME_AMOUNT = 3
+MIN_ROOM_SIZE = 1
+MAX_ROOM_SIZE = 4
+
+PERFUME_AMOUNT_SUFFIX = " Perfume Amount"
+ROOM_SIZE_SUFFIX = " Room Size"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the diffuser numbers."""
+    diffusers = hass.data[DOMAIN][config_entry.entry_id][DEVICES]
+    coordinators = hass.data[DOMAIN][config_entry.entry_id][COORDINATORS]
+    entities: list[DiffuserEntity] = []
+    for hublot, diffuser in diffusers.items():
+        coordinator = coordinators[hublot]
+        entities.append(DiffuserPerfumeAmount(diffuser, coordinator))
+        entities.append(DiffuserRoomSize(diffuser, coordinator))
+
+    async_add_entities(entities)
+
+
+class DiffuserPerfumeAmount(NumberEntity, DiffuserEntity):
```

Comment:
It's safer to start with the integration specific class in the MRO since we pass parameters to the super class.